main.cpp

```diff
@@ -220,303 +298,351 @@ struct InvoiceManager
+    {
 
     };
 
-    // 1) create an invoice - returns an Invoice struct above
-    Invoice createInvoice(std::string clientName, int invoiceNumber, float dueDate, std::string workType = "post", float workTime = 0.0f);
-    // 2) check if an invoice is late
+    Invoice createInvoice(std::string clientName, float dueDate, std::string workType = "post", float workTime = 0.0f);
     bool checkOverdue(Invoice invoice);
-    // 3) calculate outstanding balance - inputs a Client and optional invoice, returns specific balance
     float checkBalance(Client client, Invoice invoice);
 };
 
-/*
-Thing 3) Scooter rental
-5 properties:
-    1) the battery level (int)
-    2) the rental time (float)
-    3) the distance traveled (int)
-    4) the GPS coordinates (float)
-    5) the rental balance (float)
-3 things it can do:
-    1) accelerate
-    2) brake
-    3) lock
- */
+void InvoiceManager::Invoice::download(std::string format)
+{
+    std::cout << "https:\\\\my.freshbooks.com\\invoice." << format << "\n";
+}
+
+void InvoiceManager::Invoice::markAsPaid(Invoice invoiceA)
+{
+    invoiceA.totalBalance = 0.0f;
+    invoiceA.overdue = false;
+}
+
+void InvoiceManager::Invoice::duplicate(Invoice invoiceA)
+{
+    Invoice invoiceB = invoiceA;    // I would not really do this
+}
+
+InvoiceManager::Invoice InvoiceManager::createInvoice(std::string name, float date, std::string type, float time)
+{
+    Invoice newInvoice;
+    newInvoice.clientName = name;
+    numInvoices += 1;
+    newInvoice.invoiceNumber = numInvoices;
+    newInvoice.dueDate = date;
+    newInvoice.workType = type;
+    newInvoice.workTime = time;
+
+    return newInvoice;
+}
+
+bool InvoiceManager::checkOverdue(Invoice invoiceA)
+{
+    std::cout << "Invoice " << invoiceA.invoiceNumber << " is ";
+    if( invoiceA.overdue == false)
+    {
+        std::cout <<  "not ";
+    }
+    std::cout << "overdue.\n";
+
+    return invoiceA.overdue;
+}
 
 struct ScooterRental
 {
-    // 2) the rental time (float)
     float rentalTime = 0.0f;
-    // 3) the distance traveled ft (int)
     int distanceTraveled = 0;
-    // 4) the GPS coordinates (float)
     float latitude = 34.052235f;
     float longitude = -118.243683f;
-    // 5) the rental balance (float)
     float balance = 1.0f;
+    int voltage = 100;
 
-    // 1) accelerate
     void accelerate(float throttle = 0.0f);
-    // 2) brake
     void brake(float brake = 0.0f);
-    // 3) lock
-    float lock(); //returns the total cost of the trip
+    float lock();
 };
 
-/*
-Thing 4)  Paintball gun
-5 properties:
-    1) CO2 pressure (int)
-    2) paintballs in the hopper (int)
-    3) paintballs in the chamber (int)
-    4) volume of lubrication (float)
-    5) valve aperature (float)
-3 things it can do:
-    1) shoot a paintball
-    2) load a paintball into chamber
-    3) lock trigger safety
- */
+void ScooterRental::accelerate(float throttle)
+{
+    voltage = int( voltage * throttle);
+}
```

Comment:
No casting until casting is introduced later in the course
If this code produces warnings with the C-style cast, then change the variable types so it doesn't produce warnings.
- [ ] revise